include/gridtools/stencil_composition/backend_cuda/tmp_storage_sid.hpp

```diff
@@ -0,0 +1,175 @@
+                    .template set<sid::property::strides>(
+                        tmp_cuda_impl_::compute_strides<Strides, BlockSizeI, BlockSizeJ>(n_blocks_i, n_blocks_j))
+                    .template set<sid::property::ptr_diff, int_t>()
+                    .template set<sid::property::strides_kind, Strides>()));
+
+#else
+    namespace tmp_cuda_impl_ {
+        template <class Strides, class BlockSizeI, class BlockSizeJ, uint_t NColors>
+        Strides compute_strides(int_t n_blocks_i, int_t n_blocks_j) {
+            return Strides{integral_constant<int_t, 1>{}, // TODO support for default init {} in hymap
+                BlockSizeI{},
+                integral_constant<int_t, BlockSizeI{} * BlockSizeJ{}>{},
+                integral_constant<int_t, BlockSizeI{} * BlockSizeJ{} * NColors>{},
+                GT_META_CALL(meta::at_c, (Strides, 3))::value * n_blocks_i,
+                GT_META_CALL(meta::at_c, (Strides, 3))::value * n_blocks_i * n_blocks_j};
+        }
+
+        template <class BlockSizeI, class BlockSizeJ>
+        std::size_t compute_size(int_t n_blocks_i, int_t n_blocks_j, int_t k_size) {
+            return BlockSizeI{} * BlockSizeJ{} * n_blocks_i * n_blocks_j * k_size;
+        }
+
+        template <class Strides>
+        int_t compute_origin_offset(Strides const &strides, int_t shift_i, int_t shift_j) {
+            return shift_i * at_key<dim::i>(strides) + shift_j * at_key<dim::j>(strides);
+        }
+    } // namespace tmp_cuda_impl_
+
+    namespace tmp_cuda {
+        struct block_i;
+        struct block_j;
+
+        template <int_t, int_t>
+        struct blocksize {};
+    } // namespace tmp_cuda
+
+    /**
+     * @brief SID for CUDA temporaries.
+     * get_origin() points to first element of compute domain
+     * TODO(havogt): during integration we need to evaluate different types of alignment:
+     *  - no alignment (current implementation)
+     *  - alignment similar to old implementation (first data point in compute domain)
+     *  - align first element of the temporary (i.e. in the redundant computation region)
+     */
+    template <class T,
+        int_t ComputeBlockSizeI,
+        int_t ComputeBlockSizeJ,
+        int_t ExtentIMinus, // negative by convention
+        int_t ExtentIPlus,
+        int_t ExtentJMinus, // negative by convention
+        int_t ExtentJPlus,
+        uint_t NColors,
+        class Allocator,
+        class BlockSizeI = integral_constant<int_t, ComputeBlockSizeI - ExtentIMinus + ExtentIPlus>,
+        class BlockSizeJ = integral_constant<int_t, ComputeBlockSizeJ - ExtentJMinus + ExtentJPlus>,
+        class Strides = hymap::keys<dim::i, dim::j, dim::c, tmp_cuda::block_i, tmp_cuda::block_j, dim::k>::values<
+            integral_constant<int_t, 1>,
+            BlockSizeI,
+            integral_constant<int_t, BlockSizeI{} * BlockSizeJ{}>,
+            integral_constant<int_t, BlockSizeI::value * BlockSizeJ::value * NColors>,
```

Comment:
If I change `BlockSizeI::value` to `BlockSizeI{}` nvcc does a weird code replacement which produces an error.